haiku-apps/batchrename/batchrename-0.2.0.recipe

```diff
@@ -0,0 +1,49 @@
+SUMMARY="An application and Tracker add-on to rename multiple files at once"
+DESCRIPTION="Renamer is a standalone application, and Tracker add-on for Haiku. \
+You can rename multiple files in multiple directories at once, and also shows \
+a preview. It offers many renaming options, and can also rename files by contents \
+of their attributes, or the result of shell scripts in case you need that."
+HOMEPAGE="https://github.com/axeld/rename"
+COPYRIGHT="2019-2024 pinc Software"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/axeld/rename/archive/b768c8525dbaf15b055ffd0b303b8c79a17d8c9a.tar.gz"
+CHECKSUM_SHA256="d94bd208ca03f46f58ed4daef4a044810630a86df60cf1431912375116a5f229"
+SOURCE_DIR="rename-b768c8525dbaf15b055ffd0b303b8c79a17d8c9a"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+#USER_SETTINGS_FILES="settings/pinc.rename settings"
+
+PROVIDES="
+	batchrename$secondaryArchSuffix
+	cmd:rename= $portVersion
+	app:rename= $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	icu74${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	make $jobArgs BUILDHOME=`finddir B_SYSTEM_DEVELOP_DIRECTORY`
+}
+
+INSTALL()
+{
+	mkdir -p $binDir
```

Comment:
`$binDir` has the secondary architecture appended, so it is `$prefix/bin/x86` on x86, and binaries are symlinked to `$prefix/bin` with `-x86` appended.

You should be using `$prefix/bin` here instead, or append `$secondaryArchSuffix` to the `cmd:rename` PROVIDES above (the former is preferred).

Reply:
Sorry, I don't get this. As it is now (as far as I understood), the binary "rename" would end up in bin/x86, and could therefore be started with just "rename" from a Terminal, right? That's at least what I wanted to achieve here.

Reply:
If the command is in `$binDir` then the command would be `rename-x86` unless you used the `setarch x86` command and then you would be able to just use `rename`.  If you always want it to be available as `rename`, regardless of `setarch` status, then it needs to be in `$prefix/bin`

Reply:
Of course, if you're going to fix gcc2 support then you won't be needing the secondary arch and using `$binDir` is fine in that case.

Reply:
Axel is right here, the binary will end up in bin/x86 and be accessible in the path. There is no problem here.

However, for programs that don't build with gcc2, we have started to put the gcc13 version directly in bin/ anyways, as it is simpler and there is no reason to use the x86/ subdirectory in that case.

Reply:
@pulkomandy Thanks for the clarification. I wasn't aware that `bin/x86` is in the PATH by default (I don't use 32 bit much). But the extra suffixed symlink would still be unneeded.

Reply:
> @pulkomandy Thanks for the clarification. I wasn't aware that `bin/x86` is in the PATH by default (I don't use 32 bit much). But the extra suffixed symlink would still be unneeded.

It's not in my $PATH on either of my 32 bit VMs